query-engine/connector-test-kit-rs/query-engine-tests/tests/new/ref_actions/on_delete/no_action.rs

```diff
@@ -54,7 +54,7 @@ mod one2one_req {
 #[test_suite(
     suite = "noaction_onD_1to1_opt",
     schema(optional),
-    exclude(Postgres, Sqlite, MongoDb)
+    exclude(Postgres, Sqlite, MongoDb, TiDB)
```

Comment:
Why exclude TiDB here? Documentation?

Reply:
@Mini256 ?

Reply:
TiDB doesn't support foreign key for now. Related issue: https://github.com/pingcap/tidb/issues/18209

In the testcases, I enable the `relationMode='prisma'`. Deleting the parent leaves the data in a integrity-violating state. 

```
Expected error with code `P2003` and message `Foreign key constraint failed on the field`, got: `{"errors":[{"error":"Error occurred during query execution:\nInterpretationError(\"Error for binding '3'\", Some(QueryGraphBuilderError(RelationViolation(RelationViolation { relation_name: \"ChildToParent\", model_a_name: \"Child\", model_b_name: \"Parent\" }))))","user_facing_error":{"is_panic":false,"message":"The change you are trying to make would violate the required relation 'ChildToParent' between the `Child` and `Parent` models.","meta":{"relation_name":"ChildToParent","model_a_name":"Child","model_b_name":"Parent"},"error_code":"P2014"}}]}`
```

So I included TiDB in the `delete_parent_violation` testcase like MongoDB: https://github.com/prisma/prisma-engines/pull/3308/commits/802aad7edb300057453fe68da589855f4ec35d6a

Reply:
Hm. Vitess also does not support foreign keys (in the configuration we are testing), so the question is why that does not need to be excluded from these tests?

Reply:
I means that I set the`relation_mode` option to `prisma` in the `TiDBConnectorTag` settings:

https://github.com/prisma/prisma-engines/blob/802aad7edb300057453fe68da589855f4ec35d6a/query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/tidb.rs#L66

Reply:
Ah now I understand. Yes, that makes sense.

Even weirded why any special case for TiDB is needed then when Vitess is not even mentioned here - because that exactly also implies for Vitess.

We'll figure that out later.

Reply:
Also, due to the support of the foreign key feature, this test can be passed happily.